### ISSUE TITLE
Add option to sort object keys

### ISSUE THREAD

**Issue by zaim** [hapijs/qs#64]
Consider adding an option to sort object keys. This could be useful when using query strings for (or as part of) a unique key, ensuring the order of property declaration doesn't affect output.

```javascript
qs.stringify({ a: 'b', z: 'x' }) === 'a=b&z=x';
qs.stringify({ z: 'x', a: 'b' }) === 'z=x&a=b';
qs.stringify({ z: 'x', a: 'b' }, { sort: true }) === 'a=b&z=x';
```

I can work on this and provide pull request if okay.

**Comment by adam-beck** [comment]
I think a compare function would be more beneficial. Just a simple `boolean` would limit how the object keys are sorted.

**Comment by Marsup** [comment]
Doesn't mean it can't be both. It'll probably be faster without the function call.

**Comment by bobylito** [comment]
I need this, and I am to work on that today, if you don't mind.

**Comment by ljharb** [comment]
Fixed by #120
